frontend/simple/views/Join.vue

```diff
@@ -0,0 +1,157 @@
+                <table class="table is-bordered is-striped is-narrow">
+                <thead>
+                <tr>
+                  <th><i18n>Group Members</i18n></th>
+                </tr>
+                </thead>
+                <tbody>
+                <tr v-for="member in state.members" class="member">
+                  <td>
+                    <div class="media">
+                      <div class="media-left">
+                        <p class="image is-64x64">
+                          <img src="http://bulma.io/images/placeholders/128x128.png">
+                        </p>
+                      </div>
+                      <div class="media-content">
+                        <strong>{{member}}</strong>
+                      </div>
+                    </div>
+                  </td>
+                </tr>
+                </tbody>
+              </table>
+              </div>
+            </div>
+          </div>
+        <div class="column is-1"></div>
+      </div>
+    </section>
+</template>
+<script>
+import * as Events from '../../../shared/events'
+import backend from '../js/backend/'
+import * as db from '../js/database'
+export default {
+  name: 'Join',
+  mounted () {
+    this.fetchData()
+  },
+  methods: {
+    fetchData: async function () {
+      let events = await backend.eventsSince(this.$route.query.groupId, this.$route.query.groupId)
+      let [contract, ...actions] = events.map(e => {
+        return Events[e.entry.type].fromObject(e.entry, e.hash)
+      })
+      let state = contract.toVuexState()
+      actions.forEach(action => {
+        let type = action.constructor.name
+        contract.constructor.vuex.mutations[type](state, action.data)
+      })
+      this.contract = contract.data
+      this.state = state
+    },
+    accept: async function () {
+      await backend.subscribe(this.$route.query.groupId)
+      await this.$store.dispatch('synchronize', this.$route.query.groupId)
+      let latest = await db.recentHash(this.$route.query.groupId)
+      let acceptance = new Events.AcceptInvitation({ username: this.$store.state.loggedIn, inviteHash: this.$route.query.inviteHash, acceptanceDate: new Date() }, latest)
+      this.$store.commit('setCurrentGroupId', this.$route.query.groupId)
+      await backend.publishLogEntry(this.$route.query.groupId, acceptance)
+      this.$store.dispatch('deleteMail', this.$route.query.inviteHash)
+      this.$router.push({path: '/mailbox'})
+    },
+    decline: async function () {
+      let latest = await backend.latestHash(this.$route.query.groupId)
+      let declination = new Events.DeclineInvitation({ username: this.$store.state.loggedIn, inviteHash: this.$route.query.inviteHash, declinedDate: new Date() }, latest)
+      await backend.publishLogEntry(this.$route.query.groupId, declination)
+      this.$store.dispatch('deleteMail', this.$route.query.inviteHash)
+      this.$router.push({path: '/mailbox'})
+    },
```

Comment:
top-level `awaits` always need error handling, even if it's an output to the console log, so that we're prepared to update the UI accordingly with user-feedback.

Check this entire PR for more areas that need to be updated per this rule.